### PR TITLE
[Tooling] Report Firebase Test Failures as Buildkite Annotations (take 2)

### PR DESCRIPTION
**Pull request by AliSoftware**
See https://github.com/wordpress-mobile/WordPress-Android/pull/17581 for details

image

> **Warning** This PR should NOT be merged until we do a new version of the `release-toolkit` that ships with the changes from https://github.com/wordpress-mobile/release-toolkit/pull/430

### Why?

I accidentally merged https://github.com/wordpress-mobile/WordPress-Android/pull/17581 too soon, before a new release for the `release-toolkit` was done and I could point to it.

So I reverted the early merge in https://github.com/wordpress-mobile/WordPress-Android/pull/17583, and here we are in this PR, preparing the exact same changes again (that were already approved in #17581), but this time we will make sure to wait for a new release before merging it… right?

### TODO

 - [x] Point to the new version of `release-toolkit` once one gets published, then remove the Not Ready for Merge label once done.

### CODE REVIEW

**Review by ParaskP7 — APPROVED** [review]
LGTM! 💯

**Comment by AliSoftware** [comment]
>  this time we will make sure to wait for a new release before merging it… right?

Once https://github.com/Automattic/peril-settings/pull/103 lands , if someone could remove-then-re-add the "Do Not Merge" label to this PR in order to re-trigger Peril on it and ensure it will guarantee preventing accidental merge of it, that would be nice 🙂

File: Gemfile
Comment (AliSoftware) [inline Gemfile]: ⚠️ 🚧  Reminder that before merging this PR, we need to revert those temporary changes above and point to the new version of `release-toolkit` after we published it in RubyGems.

**Pull request merged**
